lib/fluent/config/literal_parser.rb

```diff
@@ -254,8 +254,15 @@ def scan_json(is_array)
               buffer << line_buffer + "\n"
               line_buffer = ""
             else
-              # '#' is a char in json string
-              line_buffer << char
+              if @ss.exist?(/\{[^}]+\}/)
```

Comment:
@daipom this check can be changed to capture the case you mentioned. So here we can make sure that `}` is not being matched with the json's ending `}`. In that way we avoid capturing the strings that dont have closing braces for interpolation

Reply:
Hmm, we would be happy to reduce the risk simply, but isn't it difficult to narrow down the exact scope of the search for the end brace?
(Don't we need to analyze JSON structure?)

Reply:
I guess only nearest thing we can do here is make sure we dont capture end braces with the interpolation braces.
TBH this problem boils down to proper parsing problem like how compilers do if we are thinking in that rigorous perspective